subprojects/docs/src/docs/release/notes.md

```diff
@@ -309,6 +309,9 @@ The following are the features that have been promoted in this Gradle release.
 
 Known issues are problems that were discovered post release that are directly related to changes made in this release.
 
+Projects with large source sets might experience out of memory errors due to changes to the [Checkstyle plugin](userguide/checkstyle_plugin.html):
```

Comment:
This needs to go into the upgrade guide as a potential breaking change. There should be no content added to this section, it is a generated list of issues only.

Reply:
Added link to upgrade guide.